enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/InternalAPIKeyAuthenticator.java

```diff
@@ -68,9 +72,20 @@ public InternalAPIKeyAuthenticator(String securityParam) {
     }
 
     @Override
+    // check if sec-websocket-protocol header is there
     public boolean canAuthenticate(RequestContext requestContext) {
-        String internalKey = requestContext.getHeaders().get(
-                ConfigHolder.getInstance().getConfig().getAuthHeader().getTestConsoleHeaderName().toLowerCase());
+        String apiType = requestContext.getMatchedAPI().getApiType();
+        String internalKey;
+        if (apiType.equalsIgnoreCase("WS")) {
+            internalKey = requestContext.getHeaders().get(
+                    ConfigHolder.getInstance().getConfig().getAuthHeader().getTestConsoleHeaderName().toLowerCase());
+            if (internalKey == null || internalKey.isBlank()) {
+                internalKey = requestContext.getHeaders().get(HttpConstants.WEBSOCKET_PROTOCOL_HEADER).split(",") [1];
+            }
+        } else  {
+            internalKey = requestContext.getHeaders().get(
+                    ConfigHolder.getInstance().getConfig().getAuthHeader().getTestConsoleHeaderName().toLowerCase());
+        }
```

Comment:
Shall we simplify this logic as follows?

```suggestion
        String apiType = requestContext.getMatchedAPI().getApiType();
        String internalKey = requestContext.getHeaders().get(
                ConfigHolder.getInstance().getConfig().getAuthHeader().getTestConsoleHeaderName().toLowerCase());
        if (apiType.equalsIgnoreCase("WS") && StringUtils.isEmpty(internalKey)) {
            internalKey = requestContext.getHeaders().get(HttpConstants.WEBSOCKET_PROTOCOL_HEADER).split(",") [1];
        }
```